python/cuml/cuml/manifold/umap.pyx

```diff
@@ -297,10 +297,20 @@ class UMAP(UniversalBase,
         'nn_descent']. 'auto' chooses to run with brute force knn if number of data rows is
         smaller than or equal to 50K. Otherwise, runs with nn descent.
     build_kwds: dict (optional, default=None)
-        Build algorithm argument {'nnd_graph_degree': 64, 'nnd_intermediate_graph_degree': 128,
-        'nnd_max_iterations': 20, 'nnd_termination_threshold': 0.0001, 'nnd_return_distances': True,
-        'nnd_n_clusters': 1}
-        Note that nnd_n_clusters > 1 will result in batch-building with NN Descent.
+        Extra options for specific build algorithms. Options for
+        ``build_algo='nn_descent'`` are:
+
+        * ``nnd_n_clusters``: The number of clusters to use. Defaults to 1. If set
+          higher, nn-descent will split the input data into ``nnd_n_clusters`` batches
+          when constructing the KNN graph. Increasing this can help reduce GPU memory
+          usage (especially if ``data_on_host=True``, at the cost of performance.
+        * ``nnd_intermediate_graph_degree``: The size of the intermediate graph.
+          If 0 (the default), will be inferred as ``1.5 * n_neighbors``.
+        * ``nnd_max_iterations``: The number of iterations to refine the KNN
+          graph for. Defaults to 20, increasing may produce a better quality
+          graph at the cost of performance.
+        * ``nnd_termination_threshold``: The termination threshold for refining
+          the KNN graph. Defaults to 0.001.
```

Comment:
Would be great to get more details on this parameter. How is the threshold calculated? cc @jinsolp